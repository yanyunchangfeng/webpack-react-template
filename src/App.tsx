import React, { FC } from 'react';
import { GlobalStyle, WrapContainer } from 'src/index.style';

const App: FC = () => {
  return (
    <>
      <GlobalStyle />
      <WrapContainer>
        <div className="container"></div>
      </WrapContainer>
    </>
  );
};

export default App;
