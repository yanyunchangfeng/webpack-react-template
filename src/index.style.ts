import styled, { createGlobalStyle } from 'styled-components';

export const GlobalStyle = createGlobalStyle`
  body { margin: 0; height: 100vh; }
  #root { height: 100%; }
`;

export const WrapContainer = styled.div`
  height: 100%;
`;
