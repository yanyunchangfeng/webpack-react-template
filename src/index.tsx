import { FC } from "react";
import { createRoot } from "react-dom/client";
import React from "react";
document.body.innerHTML = `<div id="root"></div>`;
const App: FC = () => {
  return <></>;
};
const root = createRoot(document.getElementById("root")!);
root.render(<App />);

export { App };
