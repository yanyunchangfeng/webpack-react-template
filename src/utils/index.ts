export * from './safeJsonParse';
export * from './sleep';
export * from './toggleState';
export * from './store';
