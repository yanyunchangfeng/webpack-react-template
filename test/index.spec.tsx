import { App } from "src/index";
import { create } from "react-test-renderer";
import React from "react";
describe("render App test", () => {
  it("render App correctly", () => {
    const testRenderInstance = create(<App />);
    expect(testRenderInstance.toJSON()).toMatchSnapshot();
  });
});
