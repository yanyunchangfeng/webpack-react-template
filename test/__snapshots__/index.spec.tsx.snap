// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render App test render App correctly 1`] = `null`;
